.github/workflows/development.yml

```diff
@@ -0,0 +1,46 @@
+# This workflow will do a clean install of node dependencies, build the source code and run tests across different versions of node
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-nodejs-with-github-actions
+
+name: auth0-custom-lock
+env:
+  NODE_ENV: 'development'
+  nodeVersion: '14.x'
+  pythonVersion: '3.x'
+  AUTH0_DOMAIN: 'auth-dev.mozilla.auth0.com'
+  AUTH0_CLIENT_ID: ${{ secrets.DEVELOPMENT_AUTH0_CLIENT_ID }}
+  AUTH0_CLIENT_SECRET: ${{ secrets.DEVELOPMENT_AUTH0_CLIENT_SECRET }}
+  AWS_ACCESS_KEY_ID: ${{ secrets.DEVELOPMENT_AWS_ACCESS_KEY_ID }}
+  AWS_CDN_BUCKET_NAME: ${{ secrets.DEVELOPMENT_AWS_CDN_BUCKET_NAME }}
+  AWS_SECRET_ACCESS_KEY: ${{ secrets.DEVELOPMENT_AWS_SECRET_ACCESS_KEY }}
+
+on:
+  push:
+    branches:
+      - refs/tags/*-pre
+
+jobs:
+  dev-build-and-deploy:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Check for secrets to be set
+      run: if [ ${#AWS_CDN_BUCKET_NAME} -lt 1 ]; then echo "Github Secrets not set"; exit 1; fi
+    - uses: actions/checkout@v2
+    - name: Install Node.js
+      uses: actions/setup-node@v1
+      with:
+        node-version: ${{ env.nodeVersion }}
+    - name: Build Auth0 custom lock pages
+      run: |
+        npm ci
+        npm run build
+    - name: Install Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ env.pythonVersion }}
+    - name: Install AWS CLI
+      run: pip install awscli
+    - name: Sync files to S3
+      run: aws s3 cp --recursive dist s3://$AWS_CDN_BUCKET_NAME/nlx/$GITHUB_SHA
+    - name: Deploy login page to Auth0
```

Comment:
Previously, the CI executed a test between these two steps to ensure that the file referenced by the git hash was accessible before deploying the updated NLX (to prevent deploying something broken like an NLX HTML page that references S3 hosted files that don't exist). We should probably retain that test (as though the S3 upload could return a success, it could have not transferred what we're expecting as we're just doing a `cp --recursive`).

See `sanity-checks.sh`

Reply:
It shouldn't be necessary, as:

a) this step won't execute if the copy didn't happen
b) a0deploy will error out gracefully if the file referenced by the git hash wasn't accessible

Reply:
Okay, the sanity checks have been cleaned up a ton and are smarter about their environmental variables. Lemme know if this works for you.